x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/index_migrators/rule_migrations_per_space_index_migrator.ts

```diff
@@ -71,58 +175,62 @@ export class RuleMigrationSpaceIndexMigrator {
     });
   }
 
-  private async indexMigrationDocs(docs: StoredSiemMigration[]) {
-    const indexName = this.ruleMigrationIndexAdapters.migrations.getIndexName(this.spaceId);
-    const createOperations = docs.flatMap((doc) => [
-      {
-        create: {
-          _id: doc.id,
-          _index: indexName,
-        },
-      },
-      {
-        ...doc,
-      },
-    ]);
+  /**
+   * Retrieves migration IDs from the migrations index that do not have a name field.
+   */
+  private async getMigrationIdsWithoutName(): Promise<string[]> {
+    const index = this.ruleMigrationIndexAdapters.migrations.getIndexName(this.spaceId);
 
-    return this.esClient.bulk({
-      refresh: 'wait_for',
-      operations: createOperations,
+    const result = await this.esClient.search<StoredSiemMigration>({
+      index,
+      query: { bool: { must_not: { exists: { field: 'name' } } } },
+      size: MAX_ES_SEARCH_SIZE,
+      _source: false,
     });
-  }
 
-  async run() {
-    await this.migrateRuleMigrationIndex();
+    return result.hits.hits.map(({ _id }) => {
+      assert(_id, 'document should have _id');
+      return _id;
+    });
   }
 
   /**
-   * Creates the rule migration index if it doesn't exist and indexes any missing migration documents
-   * from the rules index.
-   *
+   * Retrieves the names of all migrations from the migrations index.
+   * The names are generated based on the migration the creation order.
    */
-  private async migrateRuleMigrationIndex() {
-    const installedIndexName =
-      await this.ruleMigrationIndexAdapters.migrations.getInstalledIndexName(this.spaceId);
-    if (!installedIndexName) {
-      await this.ruleMigrationIndexAdapters.migrations.createIndex(this.spaceId);
-    }
+  private async createGetMigrationsName(): Promise<(id: string) => string> {
+    // Cache the names map to avoid repeat the aggregation query
+    if (!this.namesMap) {
+      const index = this.ruleMigrationIndexAdapters.rules.getIndexName(this.spaceId);
 
-    const existingMigrationsFromRulesIndex = await this.getExistingMigrationFromRulesIndex();
-    const existingMigrationsFromMigrationsIndex =
-      await this.getExistingMigrationFromMigrationsIndex();
+      // Same migrationIds aggregation as the getAllStats method of the data client
+      const aggregations: { migrationIds: AggregationsAggregationContainer } = {
```

Comment:
@semd, we could have reused `getExistingMigrationFromRulesIndex` , it is basically stats api.